extensions/answer_summarizers/models.py

```diff
@@ -43,42 +43,79 @@
 from core.domain import stats_domain
 
 
-def _get_hashable_value(value):
-    """This function returns a hashable version of the input value. If the
-    value itself is hashable, it simply returns that value. If it's a list, it
-    will return a tuple with all of the list's elements converted to hashable
-    types. If it's a dictionary, it will first convert it to a list of pairs,
-    where the key and value of the pair are converted to hashable types, then
-    it will convert this list as any other list would be converted.
+class _HashableAnswer(object):
```

Comment:
Suggest _HashedAnswer (you're storing the hash, and in fact equality of this object is defined by the hash, rather than an answer that happens to be hashable. The latter seems to be maintained only for reference.)

Reply:
Done.